src/v/raft/consensus.cc

```diff
@@ -1289,13 +1289,6 @@ ss::future<vote_reply> consensus::do_vote(vote_request&& r) {
         }
     }
 
-    /// set to true if the caller's log is as up to date as the recipient's
-    /// - extension on raft. see Diego's phd dissertation, section 9.6
-    /// - "Preventing disruptions when a server rejoins the cluster"
-    reply.log_ok
```

Comment:
This looks very like a recipe for a deadlock. As I remember we **must** set log_ok even when `r.term < _term`.

There is `prevote_stm` optimization. We use it to check that the  current candidate's log is up to date with the majority of the nodes before starting an election round (when the candidate isn't up to date it it doesn't make sense to inclement its term and to disrupt the current leader).

Without log_ok we can't increment term, without incrementing term we can't pass `r.term < _term`, without passing it we can't set log_ok => deadlock

Reply:
This is valid comment. Would it be possible to rework prevote request handing in a way that it will return `granted` flag based on the time since last heartbeat ? It will be independent from the term and will not cause the deadlock